testbed/HIL_Firmware/common/dac.c

```diff
@@ -0,0 +1,135 @@
+#include <stdio.h>
+#include "dac.h"
+#include "freertos/FreeRTOS.h"
+#include "esp_err.h"
+#include "driver/twai.h"
+#include "driver/spi_master.h"
+#include "driver/dac_oneshot.h"
+
+//retain 4 least significant bits
+#define V_OUT_MASK 0x0F;
+
+twai_message_t message_status = {
+    .identifier = MESSAGE_STATUS,
+    .extd = 1,
+    .data_length_code = 1,
+};
+
+//convert V_REF from mV to V
+const double STEPV12 = (V_REF/1000)/MAX_STEPS_12;
+const double STEPV8 = (V_REF/1000)/MAX_STEPS_8;
+
+static bool channel1=false;
```

Comment:
#spaces, be more descriptive with identifier. what about channel1 is this describing? Channel 1 of what?